doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -7,64 +7,179 @@ In this chapter, we introduce advanced commands to modify the way Coq
 parses and prints objects, i.e. the translations between the concrete
 and internal representations of terms and commands.
 
-The main commands to provide custom symbolic notations for terms are
-:cmd:`Notation` and :cmd:`Infix`; they will be described in the
-:ref:`next section <Notations>`. There is also a
-variant of :cmd:`Notation` which does not modify the parser; this provides a
-form of :ref:`abbreviation <Abbreviations>`. It is
-sometimes expected that the same symbolic notation has different meanings in
-different contexts; to achieve this form of overloading, Coq offers a notion
-of :ref:`notation scopes <Scopes>`.
-The main command to provide custom notations for tactics is :cmd:`Tactic Notation`.
+When inputing a term to Coq, it goes through several successive steps.
+
+* :gdef:`lexing`: the input is split in a linear stream of tokens, for
+  instance the string ``"1 + 2 * 2"`` will be split into the sequence
+  of five tokens `'1' '+' '2' '*' '2'`
+
+* :gdef:`parsing`: the previous stream is given a tree-like structure, for
+  instance here
+
+  .. code-block:: text
+     :name: after-parsing
+
+             *
+            / \
+           1   +
+              / \
+             2   2
+
+* notation :gdef:`interpretation`: each syntactic element gets translated to a
```

Comment:
Again, this is not the place to define "interpretation".  It deserves its own section, which shouldn't be in the Notations chapter.

Reply:
Well, `interpretation` is a notation specific notion so it should definitely be defined in this chapter. Maybe the definition should come with the `Notation` command? and this would just be a forward reference?